Jenkinsfile

```diff
@@ -0,0 +1,174 @@
+#!groovy
+/*
+ * This Jenkinsfile is intended to run on https://ci.jenkins-ci.org and may fail anywhere else.
+ * It makes assumptions about plugins being installed, labels mapping to nodes that can build what is needed, etc.
```

Comment:
Could make the assumptions explicit -- e.g. "needs nodes labeled `java` and `docker`" (+explanations what that means)